(#1181) Add containerd (Rootful variant) support

michalbiesek opened the pull request.

jrcheli commented:
I did some looking to see if we missed any places where we might have specified scope_getuid(), but should have used scope_geteuid().
1) src/scope.c has one of these in attach(), and another in attachCmd()
2) test/libdirtest.c has one of these in GetPathOfficial()
3) test/cfgutilstest.c has an instance where it uses getuid() but should use scope_getuid() 

I did some looking to see if we missed any places where we might have specified scope_getgid(), but should have used scope_getegid().
1) test/cfgutilstest.c has an instance where it uses getgid() but should use scope_getgid()

michalbiesek commented:
@jrcheli Let me answer Your questions:
> src/scope.c has one of these in attach(), and another in attachCmd()
I think it doesn't matter - if You insists I can change the code above to effective uid as well
> test/libdirtest.c has one of these in GetPathOfficial()
It doesn't matter but for consistency, I have added the scope_ variant there
> test/cfgutilstest.c has an instance where it uses getuid() but should use scope_getuid()
It doesn't matter in this context since this is unit test and there are no switching namespace there
> **src/nsinfo.c has one in nsInfoTranslateGid()
It doesn't matter but for consistency, I have added the effective variant there
> test/cfgutilstest.c has an instance where it uses getgid() but should use scope_getgid()
It doesn't matter but for consistency, I have added the scope_ variant there

jrcheli commented:
Sorry for the noisy review.  I appreciate you humoring me with these changes.  I agree with your take on them above, and will merge #1207 now.
